### ISSUE TITLE
AutoEncoder Training

### ISSUE THREAD

**Issue by KPHippe** [facebookresearch/all-atom-diffusion-transformer#10]
Hi All, 

Sorry for the double issue - I am trying to train the structure autoencoder according to your provided condfigs. I have built the datasets and it appears the AutoEncoder is missing loss weights for the MOF dataset. It appears the joint dataset need the MOF dataset but the VAE does not have a associated weight for the MOF sets (it appears included in the training/eval even when setting `proportion: 0.0` in the joint data config) 

Did you train the AutoEncoder with MOF data? Or should there be a MP20/QM9 only dataset that avoids MOF's getting into the training set? 

Thanks!

**Comment by chaitjo** [comment]
Thanks for your interest @KPHippe!

The configs provided in the repo by default correspond to the models trained on QM9 + MP20 (without QMOF) as this is the setting that most of our results in the paper are also reported with. To enable QMOF, you can simply change the proportion of QMOF to not be 0. 

If you want the QMOF version of the model, you should obviously also train with QMOF.
 
The QMOF part of the code may have some minor semantic changes I need to make. I plan to make the necessary fixes and create a pull request as soon as I have time.

**Comment by Colin-Jay** [comment]
@KPHippe hi, I believe your issue can be easily resolved by adding the following code after https://github.com/facebookresearch/all-atom-diffusion-transformer/blob/main/src/data/joint_datamodule.py#L173, if you don't need the QMOF data. 

```python
self.qmof150_train_dataset = self.qmof150_train_dataset[
    : int(len(self.qmof150_train_dataset) * self.hparams.datasets.qmof150.proportion)
]
self.qmof150_val_dataset = self.qmof150_val_dataset[
    : int(len(self.qmof150_val_dataset) * self.hparams.datasets.qmof150.proportion)
]
self.qmof150_test_dataset = self.qmof150_test_dataset[
    : int(len(self.qmof150_test_dataset) * self.hparams.datasets.qmof150.proportion)
]

**Comment by Colin-Jay** [comment]
> Hi All,
> 
> Sorry for the double issue - I am trying to train the structure autoencoder according to your provided condfigs. I have built the datasets and it appears the AutoEncoder is missing loss weights for the MOF dataset. It appears the joint dataset need the MOF dataset but the VAE does not have a associated weight for the MOF sets (it appears included in the training/eval even when setting `proportion: 0.0` in the joint data config)
> 
> Did you train the AutoEncoder with MOF data? Or should there be a MP20/QM9 only dataset that avoids MOF's getting into the training set?
> 
> Thanks!

@KPHippe hi, I believe your issue can be easily resolved by adding the following code after https://github.com/facebookresearch/all-atom-diffusion-transformer/blob/main/src/data/joint_datamodule.py#L173, if you don't need the QMOF data. 

```python
self.qmof150_train_dataset = self.qmof150_train_dataset[
    : int(len(self.qmof150_train_dataset) * self.hparams.datasets.qmof150.proportion)
]
self.qmof150_val_dataset = self.qmof150_val_dataset[
    : int(len(self.qmof150_val_dataset) * self.hparams.datasets.qmof150.proportion)
]
self.qmof150_test_dataset = self.qmof150_test_dataset[
    : int(len(self.qmof150_test_dataset) * self.hparams.datasets.qmof150.proportion)
]

**Comment by KPHippe** [comment]
Great! I added this as well, but thought it was worth mentioning there is a bit of confusion with some parameters (maybe just my issues 😄)

**Comment by chaitjo** [comment]
Thanks for finding this! Sorry for some of these pesky/obvious bugs as I didn't have access to the original cluster that this codebase was developed for when compiling and collating it.

**Comment by chaitjo** [comment]
You may also have to reduce batch size per GPU when using QMOF data - depends on your GPU's RAM.

**Comment by chaitjo** [comment]
https://github.com/chaitjo/all-atom-diffusion-transformer/tree/dev

A more up-to-date repo with fixes based on this issue is under development. We'll merge it into this main repo soon. Datasets now auto-download from huggingface.
